tests/eval_files/async121.py

```diff
@@ -0,0 +1,97 @@
+# ASYNCIO_NO_ERROR # checked in async121_asyncio.py
+# ANYIO_NO_ERROR # checked in async121_anyio.py
+
+import trio
+
+
+def condition() -> bool:
+    return False
+
+
+async def foo_return():
+    async with trio.open_nursery():
+        if condition():
+            return  # ASYNC121: 12, "return", "nursery"
+        while condition():
+            return  # ASYNC121: 12, "return", "nursery"
+
+    return  # safe
+
+
+async def foo_return_nested():
+    async with trio.open_nursery():
+
+        def bar():
+            return  # safe
+
+
+# continue
+async def foo_while_continue_safe():
+    async with trio.open_nursery():
+        while True:
+            continue  # safe
+
+
+async def foo_while_continue_unsafe():
+    while True:
+        async with trio.open_nursery():
+            if condition():
+                continue  # ASYNC121: 16, "continue", "nursery"
+        continue  # safe
```

Comment:
What happens if we have a loop, containing a condition, containing `continue` or `break`?  Let's test that?

Reply:
The if statements are solely for avoiding mypy unreachable-statement, and should have zero effect on the visitor logic. But think I added what you requested?